### PR TITLE
[refactor] split out some of the image related lowering into its own …

### PR DESCRIPTION
**Pull request by schweitzpgi**
…file

### CODE REVIEW

File: flang/include/flang/Lower/Image.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
+//===-- Lower/Image.h -- image related lowering -----------------*- C++ -*-===//
```
Comment (jeanPerier) [inline flang/include/flang/Lower/Image.h:1]: Would Coarray.h/.cpp fit for these files ? Image.h is fine but Coarray.h may be help to grasp faster what this is about.

File: flang/lib/Lower/Image.cpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    Fortran::lower::pft::Evaluation &,
+    const Fortran::parser::EndChangeTeamStmt &) {
+  TODO("END CHANGE TEAM");
+}
+
+void Fortran::lower::genFormTeamStatement(
+    Fortran::lower::AbstractConverter &converter,
+    Fortran::lower::pft::Evaluation &, const Fortran::parser::FormTeamStmt &) {
+  // FIXME: There is no runtime call to make for this yet.
+  noRuntimeSupport(converter.getCurrentLocation(), "FORM TEAM");
```
Comment (jeanPerier) [inline flang/lib/Lower/Image.cpp:62]: Would `TODO("FORM TEAM stmt");` also work here so `noRuntimeSupport` can be deleted ? (I am not sure its only a lack of runtime support, there might be more to do here)

Comment (schweitzpgi) [reply flang/lib/Lower/Image.cpp:62]: Sure. I just moved that code over.

File: flang/include/flang/Lower/Image.h (hunk 1 above)
Comment (schweitzpgi) [reply flang/include/flang/Lower/Image.h:1]: I can rename them, but I thought "all things image related" was more general than simply "coarray".

**Comment by sscalpone** [comment]
I don't know how this came out, but from the Fortran reference:
`3.84 image - instance of a Fortran program`
and
`3.27 coarray - data entity that has nonzero corank`
So coarray is most descriptive.

**Pull request merged**